smartseq2_single_sample/run_multi_samples/ss2_single_sample_wf.wdl

```diff
@@ -0,0 +1,300 @@
+task Star {
+  File input_fastq_read1
+  File input_fastq_read2
+  File gtf
+  File star_genome
+  String sample_tag
+  String pu_tag
+  String id_tag
+  String lib_tag
+
+
+  command {
+    tar -xvf ${star_genome}
+    STAR  --readFilesIn ${input_fastq_read1} ${input_fastq_read2} \
+      --genomeDir ./star \
+      --quantMode TranscriptomeSAM \
+      --outSAMstrandField intronMotif \
+      --runThreadN 8 \
+      --genomeLoad NoSharedMemory \
+      --sjdbGTFfile ${gtf} \
+      --readFilesCommand "zcat" \
+      --twopassMode Basic \
+      --outSAMtype BAM SortedByCoordinate  \
+      --outSAMunmapped Within \
+      --limitBAMsortRAM 30000000000 \
+      --outSAMattrRGline ID:${id_tag} PL:illumina PU:${pu_tag} SM:${sample_tag} LB:${lib_tag}
+  }
+  output {
+    File junction_table = "SJ.out.tab"
+    File output_bam = "Aligned.sortedByCoord.out.bam"
+    File output_bam_trans = "Aligned.toTranscriptome.out.bam"
+  }
+  runtime {
+    docker:"humancellatlas/star_dev:v1"
+    memory: "40 GB"
+    disks :"local-disk 100 HDD"
+    cpu: "8"
+  }
+}
+
+task FeatureCountsUniqueMapping {
+  File aligned_bam
+  File gtf
+  String fc_out
+  
+  command {
+    featureCounts -s 0 -t exon -g gene_id -p -B -C -a ${gtf} -o "${fc_out}.gene.unq.counts.txt" ${aligned_bam}
+    featureCounts  -s 0 -t exon -g transcript_id -p -B -C -a ${gtf} -o "${fc_out}.transcript.unq.counts.txt" ${aligned_bam}
+    featureCounts -s 0 -t exon -g exon_id -p -B -C -a ${gtf} -o "${fc_out}.exon.unq.counts.txt" ${aligned_bam}
+  }
+  runtime {
+    docker:"humancellatlas/star_dev:v1"
+    memory: "15 GB"
+    disks: "local-disk 50 HDD"
+  }
+  output {
+    File genes = "${fc_out}.gene.unq.counts.txt"
+    File exons = "${fc_out}.exon.unq.counts.txt"
+    File trans = "${fc_out}.transcript.unq.counts.txt"
+  }
+}
+
+task FeatureCountsMultiMapping {
+  File aligned_bam
+  File gtf
+  String fc_out
+
+  command {
+    featureCounts  -s 0 -t exon -g gene_id -p -M -O -a ${gtf} -o "${fc_out}.gene.mult.counts.txt" ${aligned_bam}
+    featureCounts  -s 0 -t exon -g transcript_id -p -M -O -a ${gtf} -o "${fc_out}.transcript.mult.counts.txt" ${aligned_bam}
+    featureCounts  -s 0 -t exon -g exon_id -p -M  -O -a ${gtf} -o "${fc_out}.exon.mult.counts.txt" ${aligned_bam}
+  }
+  runtime {
+    docker: "humancellatlas/star_dev:v1"
+    memory: "15 GB"
```

Comment:
15G ram nets you 4 processors. Could you run featureCounts concurrently by backgrounding the first 2 processes? would that increase the ram cost? 

Alternatively, it looks like these tasks aren't dependent upon each other. Could this task be scattered across `exon_id`, `gene_id`, and `transcript_id`?